pkg/collector/corechecks/orchestrator/pod/pod.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubelet && orchestrator
+
+package pod
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.uber.org/atomic"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors"
+	k8sProcessors "github.com/DataDog/datadog-agent/pkg/collector/corechecks/cluster/orchestrator/processors/k8s"
+	"github.com/DataDog/datadog-agent/pkg/orchestrator"
+	oconfig "github.com/DataDog/datadog-agent/pkg/orchestrator/config"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/clustername"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+)
+
+const checkName = "pod"
+
+var groupID atomic.Int32
+
+func nextGroupID() int32 {
+	groupID.Add(1)
+	return groupID.Load()
+}
+
+func init() {
+	core.RegisterCheck(checkName, PodFactory)
+}
+
+// Check doesn't need additional fields
+type Check struct {
+	core.CheckBase
+	hostName  string
+	clusterID string
+	sender    sender.Sender
+	processor *processors.Processor
+	config    *oconfig.OrchestratorConfig
+}
+
+// PodFactory returns a new Pod.Check
+func PodFactory() check.Check {
+	return &Check{
+		CheckBase: core.NewCheckBase(checkName),
+		config:    oconfig.NewDefaultOrchestratorConfig(),
+	}
+}
+
+// Configure the CPU check
+// nil check to allow for overrides
+func (c *Check) Configure(
+	senderManager sender.SenderManager,
+	integrationConfigDigest uint64,
+	data integration.Data,
+	initConfig integration.Data,
+	source string,
+) error {
+	c.BuildID(integrationConfigDigest, data, initConfig)
+
+	err := c.CommonConfigure(senderManager, integrationConfigDigest, initConfig, data, source)
+	if err != nil {
+		return err
+	}
+
+	err = c.config.Load()
+	if err != nil {
+		return err
+	}
+
+	if !c.config.OrchestrationCollectionEnabled {
+		return errors.New("orchestrator check is configured but the feature is disabled")
+	}
+	if !c.config.CoreCheck {
+		return errors.New("the corecheck version for pods is currently disabled")
```

Comment:
This follows the patterns of some of our other checks. I'm currently working on a cutover plan which will give users more warning. However, is there a better way to not schedule the check if say `orchestrator_explorer.enabled` = false vs erroring?

Reply:
I think [staticconfig](https://github.com/DataDog/datadog-agent/blob/4277ccc6a8ac5a9cc5d58cf5c5c91a0daacc26d7/pkg/autodiscovery/listeners/staticconfig.go#L49-L59) configures checks based on config flags. Alternatively, you could make the check opt-in, or maybe it could would fit as a part of the [kubelet check](https://github.com/DataDog/datadog-agent/blob/4277ccc6a8ac5a9cc5d58cf5c5c91a0daacc26d7/pkg/collector/corechecks/containers/kubelet/kubelet.go#L71-L76) which already collects data from kubernetes. Or the check could be opt-in instead of being enabled by default.

Reply:
I spoke with @sblumenthal and we think it's best to keep the check separate from kubelet for now. I'll check out staticconfig. I'm still looking into the transition plan, but will make sure we do it in a way that gives customers time and warning before migration